File: Futbol5/Futbol5.Web/Scripts/app/services/equipo.service.js
(function () {
    'use strict';
    angular.module('app').factory('equipoService', equipoService);

    equipoService.$inject = ['$resource', 'serviceHelper'];

    function equipoService($resource, serviceHelper) {

        var service = {
            crearEquipo,
            eliminarEquipo,
            modificarEquipo,
            obtenerEquipoPorId,
            obtenerEquipos
        };

        var equipo = serviceHelper.Equipo;

        function crearEquipo(data) {
            return equipo.crearEquipo(data);
        };

        function eliminarEquipo(id) {
            return equipo.eliminarEquipo({ equipoId: id });
        };

        function modificarEquipo(data) {
            return equipo.modificarEquipo(data);
        };

        function obtenerEquipoPorId(id) {
            return equipo.obtenerEquipoPorId({ id: id });
        };

        function obtenerEquipos() {
            return equipo.obtenerEquipos();
        };

        return service;
    };
})();

File: Futbol5/Futbol5.ServiceAgent/EquipoService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Futbol5.DAL.Repositories;
using Futbol5.DAL.Infrastructure;
using Futbol5.Entities;

namespace Futbol5.ServiceAgent
{
    public class EquipoService : IEquipoService
    {
        private IEquipoRepository repository;
        private IUnitOfWork unitOfWork;

        public EquipoService(IEquipoRepository repository, IUnitOfWork unitOfWork)
        {
            this.repository = repository;
            this.unitOfWork = unitOfWork;
        }

        public IEnumerable<Equipo> ObtenerEquipos()
        {
            return repository.GetAll();
        }

        public Equipo AgregarEquipo(Equipo equipo)
        {
            repository.Add(equipo);
            SaveChanges();

            return equipo;
        }

        public Equipo ObtenerEquipoPorId(int id)
        {
            return repository.GetById(id);
        }

        public Equipo ModificarEquipo(Equipo equipo)
        {
            repository.Update(equipo);
            SaveChanges();

            return equipo;
        }

        public void EliminarEquipo(int id)
        {
            var equipo = repository.GetById(id);
            repository.Delete(equipo);

            SaveChanges();
        }

        public void SaveChanges()
        {
            unitOfWork.SaveChanges();
        }
    }

    public interface IEquipoService
    {
        IEnumerable<Equipo> ObtenerEquipos();
        Equipo AgregarEquipo(Equipo equipo);
        Equipo ObtenerEquipoPorId(int id);
        Equipo ModificarEquipo(Equipo equipo);
        void EliminarEquipo(int equipoId);
    }
}
